yt_dlp/extractor/ciscolive.py

```diff
@@ -2,144 +2,179 @@
-        'only_matching': True,
+        'url': 'https://www.ciscolive.com/on-demand/on-demand-library.html?search.event=1707169032930001EEu2&search.technology=1538390420915002wPJx#/',
+        'info_dict': {
+            'id': 'search',
+        },
+        'playlist_count': 34,
     }]
 
     @classmethod
     def suitable(cls, url):
         return False if CiscoLiveSessionIE.suitable(url) else super().suitable(url)
 
-    @staticmethod
-    def _check_bc_id_exists(rf_item):
-        return int_or_none(try_get(rf_item, lambda x: x['videos'][0]['url'])) is not None
-
-    def _entries(self, query, url):
-        query['size'] = 50
-        query['from'] = 0
-        for page_num in itertools.count(1):
-            results = self._call_api(
-                'search', None, query, url,
-                f'Downloading search JSON page {page_num}')
-            sl = try_get(results, lambda x: x['sectionList'][0], dict)
-            if sl:
-                results = sl
-            items = results.get('items')
-            if not items or not isinstance(items, list):
+    def _entries(self, payload):
+        from_val = 0
+
+        for page in itertools.count(1):
+            search = self._call_api(
+                'search', None, {
+                    **payload,
+                    'from': from_val,
+                }, True, note=f'Downloading page {page}')
+            if not traverse_obj(search, 'sectionList'):
+                return
+
+            yield from [self.url_result(
+                f'{self._BASE_URL}/on-demand/on-demand-library.html#/session/{session_id}', CiscoLiveSessionIE)
+                for session_id in traverse_obj(search, (
+                    'sectionList', ..., 'items', ..., 'sessionID', {str_or_none}, filter, all, filter))]
+
+            from_val += int_or_none(payload['size'])
+            if from_val >= min(500, traverse_obj(search, (
+                'sectionList', ..., 'total', {int_or_none}, any,
+            ))):
                 break
-            for item in items:
-                if not isinstance(item, dict):
-                    continue
-                if not self._check_bc_id_exists(item):
-                    continue
-                yield self._parse_rf_item(item)
-            size = int_or_none(results.get('size'))
-            if size is not None:
-                query['size'] = size
-            total = int_or_none(results.get('total'))
-            if total is not None and query['from'] + query['size'] > total:
-                break
-            query['from'] += query['size']
 
     def _real_extract(self, url):
-        query = parse_qs(url)
-        query['type'] = 'session'
-        return self.playlist_result(
-            self._entries(query, url), playlist_title='Search query')
+        payload = {
+            'size': '50',
+            'type': 'session',
+            **parse_qs(url),
+        }
+
+        return self.playlist_result(self._entries(payload), 'search')
```

Comment:
Instead of `'search'`, maybe use the query parameters from the URL as the ID; if the order isn't significant, `urlencode_postdata(sorted(parse_qs(url).items())`